### PR TITLE
MNT Fix bug: FloatTransformer output dimension was set incorrectly.

### PR DESCRIPTION
**Pull request by SeanMcCarren**
## Description
Thanks to [this elaborately typed out issue](https://github.com/fairlearn/fairlearn/issues/1332) we found a bug in Adversarial Debiasing. Considering how big, highly parametrized, and not fully covered this implementation is, I'm surprised this is the first bug :)

The bug is simple. The `FloatTransformer` class's intention is to cast different types of input to a 2D array. But, in a specific case where no transformer is provided as input, and the output is not bi/multinomial, we accidentally set the output shape as the input shape.

## Tests

- [ ] no new tests required
- [ ] new tests added
- [x] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

**Comment by hildeweerts** [comment]
Many thanks, @SeanMcCarren! This looks good to me - but it seems the tests are failing unrelated to your PR (see #1334). After we fix that and the PR passes all checks I think it should be ready for merging.

File: fairlearn/adversarial/_preprocessor.py [hunk 1]
```diff
@@ -103,11 +103,16 @@ def _check(self, X, dtype=None, init=False):
...
+            if self.input_dim_ > 2:
+                raise ValueError(
+                    "Data can be at most two dimensional, not %d" % self.input_dim_
+                )
         else:
             if X.ndim != self.input_dim_:
                 raise ValueError("Dimension of data is inconsistent with previous call")
         if X.ndim == 1:
             X = X.reshape(-1, 1)
+        assert X.ndim == 2
```
Comment (romanlutz) [inline fairlearn/adversarial/_preprocessor.py:115]: An if check with raising an exception with a nice error message is probably preferable. Unless this is a case that can't happen at all?

Comment (SeanMcCarren) [reply fairlearn/adversarial/_preprocessor.py:115]: It really shouldn't happen, but I agree!

**Review by romanlutz — APPROVED** [review]

**Comment by joelthe1** [comment]
Please consider merging and releasing this fix. Thanks!

File: fairlearn/adversarial/_preprocessor.py [hunk 2]
```diff
@@ -103,11 +103,17 @@ def _check(self, X, dtype=None, init=False):
         )
         if init:
             self.input_dim_ = X.ndim
+            if self.input_dim_ > 2:
+                raise ValueError(
```
Comment (riedgar-ms) [inline fairlearn/adversarial/_preprocessor.py:107]: Sorry for the delay in reviewing, but is it possible to trigger this error in a test? Same for the `SystemError` below?

**Comment by SeanMcCarren** [comment]
I am surprised this isn't merged yet, I am completely out of the loop of this. I can confirm that the issue raised [here](https://github.com/fairlearn/fairlearn/issues/1360) by [parasurama](https://github.com/parasurama) will also be fixed by this PR.

**Review by riedgar-ms — APPROVED** [review]

**Pull request merged**